docker/zenml-server-dev.Dockerfile

```diff
@@ -95,37 +124,37 @@ ENV \
   # Set the ZenML server login rate limit to 100 requests per minute
   ZENML_SERVER_LOGIN_RATE_LIMIT_MINUTE=100
 
-WORKDIR /zenml
-
 # Install runtime dependencies
 #
 # NOTE: System packages required at runtime should be installed here
 
-# Install some utilities for debugging and development
-RUN set -ex \
-  && apt-get update \
-  && apt-get install -y curl net-tools nmap inetutils-ping default-mysql-client mariadb-client git \
-  && apt-get clean -y \
-  && rm -rf /var/lib/apt/lists/*
-
-# Copy the virtual environment from the builder stage
-COPY --from=builder /opt/venv /opt/venv
+# Copy the virtual environment with all dependencies
+COPY --chown=$USERNAME:$USER_GID --from=builder /opt/venv /opt/venv
 # Copy the requirements.txt file from the builder stage
-COPY --from=builder /zenml/requirements.txt /zenml/requirements.txt
+COPY --chown=$USERNAME:$USER_GID --from=builder /zenml/requirements.txt /zenml/requirements.txt
 
-# Create the user and group which will be used to run the ZenML server
-# and set the ownership of the workdir directory to the user.
-# Create the local stores directory beforehand and ensure it is owned by the
-# user.
-RUN groupadd --gid $USER_GID $USERNAME \
-    && useradd --uid $USER_UID --gid $USER_GID -m $USERNAME \
-    && mkdir -p /zenml/.zenconfig/local_stores/default_zen_store \
-    && chown -R $USER_UID:$USER_GID /zenml
+# Copy source code
+COPY --chown=$USERNAME:$USER_GID README.md pyproject.toml ./
+COPY --chown=$USERNAME:$USER_GID src src
 
-ENV PATH="$VIRTUAL_ENV/bin:/home/$USERNAME/.local/bin:$PATH"
+FROM common-runtime AS local-runtime
```

Comment:
This local runtime build stage is the one that I would use when mounting my local files for easier development with the docker compose setup, correct?

Reply:
Correct !

Reply:
It's all covered in the docker-compose file.